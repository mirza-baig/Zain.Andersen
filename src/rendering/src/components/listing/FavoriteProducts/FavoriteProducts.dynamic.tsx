// Global
import { useState, useEffect, MouseEvent } from 'react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck, Text, useSitecoreContext } from '@sitecore-jss/sitecore-jss-nextjs';
import Link from 'next/link';
import classNames from 'classnames';

// Components
import { Component } from 'src/helpers/Component';
import { FavoriteProductsTheme } from './FavoriteProducts.theme';
import { Headline } from 'src/helpers/Headline';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import { ImagePrimary, Image } from 'src/helpers/Media';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import { NoResults } from './NoResults.helper';
import { Spinner } from 'src/helpers/Spinner';
import { useBVScript } from 'lib/utils/use-bv-script';

type priceLevel = 1 | 2 | 3 | 4 | 5;

export type FavoriteProductsProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.Listing.Favorites.FavoriteProducts;
const FavoriteProducts = (props: FavoriteProductsProps) => {
  const { fields } = props;

  const sitecoreContext = useSitecoreContext();
  const language = sitecoreContext.sitecoreContext.language ?? 'en';

  const { themeData } = useTheme(FavoriteProductsTheme);
  const [favoriteProductDetailsArray, setFavoriteProductDetailsArray] = useState([]);

  // fetcher function for SWR
  const fetcher = async (url: string, favoriteProducts: string[], language: string) => {
    const response = await fetch(url, {
      method: 'POST',
      body: JSON.stringify({ favoriteProducts, language }),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (!response.ok) {
      throw new Error('Network response was not ok');
    }

    return response.json();
  };

  const [productData] = useState([]);
  const [error, setError] = useState(null);

  const [favProductIds, setFavProductIds] = useState(
    JSON.parse(localStorage.getItem('aw_favorites_products') || '[]')
  );
  const hasFavoriteProduct = favProductIds && favProductIds.length ? true : false;

  useEffect(() => {
    if (favProductIds && favProductIds.length > 0) {
      fetcher('/api/enterprise/favorite-products/', favProductIds, language)
        .then((data) => {
          setFavoriteProductDetailsArray(data.productData);
        })
        .catch((err) => setError(err));
    } else {
      setFavoriteProductDetailsArray([]);
    }
    // "language" is comig from sitecore context which is not going to change without page refreshing.
    // We can ignore the warning for this useEffect.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [favProductIds]);

  // Add the bazaarvoice script
  useBVScript();

  const handlePrint = () => {
    const printStyles = `
    @media print {
      header,
      footer,
      .react-tabs__tab-list,
      .no-print,
      section[data-component="hero/herotwocolumn"],
      section[data-component="hero/tabsgeneralcontent"] h2 {
        display: none;
      }

      .printSection {
        display: inherit !important;
      }
      .printOnly {
        display: block !important;
      }
    }
  `;

    const style = document.createElement('style');
    style.innerHTML = printStyles;

    document.head.appendChild(style);
    window.print();
    document.head.removeChild(style);
  };

  const priceRange = (priceLevel: priceLevel) => {
    switch (priceLevel) {
      case 1:
        return (
          <>
            <Text
              tag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$', value: '$' }}
            />
            <Text
              tag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$$$$', value: '$$$$' }}
            />
          </>
        );
      case 2:
        return (
          <>
            <Text
              tag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$$', value: '$$' }}
            />
            <Text
              ttag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$$$', value: '$$$' }}
            />
          </>
        );
      case 3:
        return (
          <>
            <Text
              tag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$$$', value: '$$$' }}
            />
            <Text
              tag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$$', value: '$$' }}
            />
          </>
        );
      case 4:
        return (
          <>
            <Text
              tag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$$$$', value: '$$$$' }}
            />
            <Text
              ttag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$', value: '$' }}
            />
          </>
        );
      case 5:
        return (
          <>
            <Text
              tag="span"
              className={themeData.classes.priceLevel}
              field={{ editable: '$$$$$', value: '$$$$$' }}
            />
          </>
        );
      default:
        return null;
    }
  };

  // As of now, we don't have any type or interface defined for favoriteProductDetailsArray or favProduct
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const ColorAndActions = ({ favProduct }: any): JSX.Element => {
    const colorSwatchesAll = favProduct.standardInteriorColors?.targetItem?.swatches?.colors;
    const colorSwatchesTotal = colorSwatchesAll && colorSwatchesAll.length;

    let featuredInteriorColors = favProduct.featuredInteriorColors.colors || [];

    let colorSwatchesCount = 0;

    if (colorSwatchesTotal && featuredInteriorColors?.length > 0) {
      featuredInteriorColors = featuredInteriorColors.slice(0, 3);
      colorSwatchesCount = colorSwatchesTotal - featuredInteriorColors.length;
    }

    return (
      <>
        {featuredInteriorColors?.length > 0 && (
          <div className={themeData.classes.colorSwatchesWrapper}>
            <Text tag={'h4'} field={{ value: 'Colors' }} className={themeData.classes.colorLabel} />
            <Link href={favProduct.productDetailPageLink.url ?? ''}>
              <a>
                <div className={themeData.classes.swatches}>
                  {featuredInteriorColors?.map(
                    // As of now, we don't have any type or interface defined for favoriteProductDetail item and its properties (featuredInteriorColors items in this case)
                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
                    (color: any, index: number) => {
                      return (
                        <div key={index} className={themeData.classes.colorSwatches}>
                          <Image image={{ value: color.swatchImage }} layout="intrinsic" />
                        </div>
                      );
                    }
                  )}
                  {colorSwatchesCount > 0 && <div>+{colorSwatchesCount}</div>}
                </div>
              </a>
            </Link>
          </div>
        )}
        {!!favProduct.productDetailPageLink.text && (
          <div className={themeData.classes.actions}>
            <SingleButton
              classes={
                (themeData.classes.buttonGroupClass,
                { wrapper: 'mb-0 md:mb-0', cta1Classes: 'font-bold' })
              }
              fields={{
                cta1Link: {
                  value: {
                    href: favProduct.productDetailPageLink.url,
                    text: favProduct.productDetailPageLink.text,
                    anchor: favProduct.productDetailPageLink.anchor,
                    target: favProduct.productDetailPageLink.target,
                  },
                },
                cta1AriaLabel: {
                  value: '',
                },
                cta1ModalLinkText: {
                  value: '',
                },
                cta1Style: {
                  id: '',
                  url: '',
                  name: 'Primary',
                  displayName: 'Primary',
                  fields: {
                    Value: {
                      value: 'primary',
                    },
                  },
                  templateId: '',
                  templateName: 'Enum',
                },
                cta1Icon: {
                  id: '',
                  url: '',
                  name: 'Arrow',
                  displayName: 'Arrow',
                  fields: {
                    Value: {
                      value: 'arrow',
                    },
                  },
                  templateId: '',
                  templateName: 'Enum',
                },
              }}
            />
          </div>
        )}
      </>
    );
  };

  const handleFavoriteProductClick = (event: MouseEvent) => {
    const target = event.target as Element;
    const link = target && target.closest('.favorite-product-item');
    if (!link) {
      return;
    }

    const productID = link.getAttribute('data-product-id');
    const favoritesProducts = JSON.parse(localStorage.getItem('aw_favorites_products') || '[]');

    const existingProductIndex = favoritesProducts.findIndex(
      (obj: string | null) => obj === productID
    );

    if (existingProductIndex !== -1) {
      favoritesProducts.splice(existingProductIndex, 1);
      link.classList.remove('favorited', 'border-[transparent_#f26924_transparent_transparent]');
    } else {
      favoritesProducts.unshift(productID);
      link.classList.add('favorited', 'border-[transparent_#f26924_transparent_transparent]');
    }

    if (favoritesProducts.length === 0) {
      localStorage.removeItem('aw_favorites_products');
      localStorage.setItem('aw_favorites_products_count', '0');

      const countStorageEvent = new StorageEvent('storage', {
        key: 'aw_favorites_products_count',
        newValue: '0',
      });
      // Dispatch the event
      window.dispatchEvent(countStorageEvent);
    } else {
      localStorage.removeItem('aw_favorites_products_count');
      localStorage.setItem('aw_favorites_products', JSON.stringify(favoritesProducts));

      // Trigger the storage event to update count across tabs/windows
      const storageEvent = new StorageEvent('storage', {
        key: 'aw_favorites_products',
        newValue: JSON.stringify(favoritesProducts),
      });
      window.dispatchEvent(storageEvent);
    }

    const remainingFavoriteItems = JSON.parse(
      localStorage.getItem('aw_favorites_products') || '[]'
    );
    setFavProductIds(remainingFavoriteItems);
  };

  return (
    <Component variant="lg" dataComponent="listing/favoriteproducts" {...props}>
      <div className="printSection col-span-12 flex justify-between">
        <Headline useTag="h4" classes={themeData.classes.mainheadline} {...props} />
        {productData?.length > 0 && hasFavoriteProduct && !error && (
          <div
            onClick={handlePrint}
            className="no-print flex cursor-pointer items-center text-body text-primary max-md:hidden"
          >
            <SvgIcon icon="pdf-aw" className="mr-xxs" />
            Print
          </div>
        )}
      </div>
      <div className="col-span-12">
        {hasFavoriteProduct ? (
          // Favorite Products
          <>
            {!productData && !error && (
              <div className="loader flex min-h-[40vh] w-full items-center justify-center">
                <Spinner size={48} />
              </div>
            )}

            {error && (
              <div className="font-sans text-sm-m font-medium md:text-m">
                {
                  "We're sorry, we couldn't process your request at this time. Please refresh or try again later."
                }
              </div>
            )}
            <div className="printSection grid grid-cols-1 gap-4 ml:grid-cols-3">
              {!error &&
                favoriteProductDetailsArray.length > 0 &&
                favoriteProductDetailsArray.map(
                  // As of now, we don't have any type or interface defined for favoriteProductDetailsArray or its items
                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
                  (favProduct: any, index: number) => {
                    const productID = favProduct.productId.value;
                    const isFavorited = favProductIds.includes(productID);
                    const priceLevel =
                      favProduct.priceLevel.targetItem &&
                      favProduct.priceLevel.targetItem.priceLevelText.value &&
                      (+favProduct.priceLevel.targetItem.priceLevelText.value as priceLevel | null);
                    const bazaarvoiceProductId = favProduct?.bazaarvoiceProductId?.value;

                    return (
                      <div key={index} className="mb-m">
                        <div className={themeData.classes.productPreviewCradWrapper}>
                          {/* Favourite */}
                          <div className="absolute top-0 right-0">
                            <div
                              className={classNames(
                                themeData.classes.favoriteProduct,
                                isFavorited
                                  ? ' favorite-product-item favorited border-[transparent_#f26924_transparent_transparent]'
                                  : ''
                              )}
                              data-product-id={favProduct.productId.value}
                              onClick={handleFavoriteProductClick}
                            >
                              <SvgIcon
                                icon="favorite"
                                fillId="white"
                                size="xl"
                                className={themeData.classes.favoriteIcon}
                              />
                            </div>
                          </div>

                          <div className={themeData.classes.headerWrapper}>
                            <Text
                              tag="h4"
                              className={themeData.classes.eyebrow}
                              field={{
                                value: favProduct.productSeries?.targetItem?.productTypeName?.value,
                              }}
                            />
                            <Link href={favProduct.productDetailPageLink.url ?? ''}>
                              <a className={themeData.classes.headlineWrapper}>
                                <Text
                                  useTag="h4"
                                  className={themeData.classes.headline}
                                  field={{ value: favProduct.productName.value }}
                                />
                              </a>
                            </Link>

                            {/* This is a placeholder for the ratings/price API to be configured */}
                            <div className={themeData.classes.ratingsAndPriceWrapper}>
                              {bazaarvoiceProductId && (
                                <div
                                  className="review flex-[0_0_auto]"
                                  data-bv-show="inline_rating"
                                  data-bv-product-id={bazaarvoiceProductId}
                                  data-bv-seo="false"
                                ></div>
                              )}

                              {/* Price Range */}
                              {priceLevel && priceLevel > 0 && (
                                <div
                                  className={
                                    bazaarvoiceProductId ? themeData.classes.priceLevelWrapper : ''
                                  }
                                >
                                  <div className={themeData.classes.priceText}>
                                    {priceRange(priceLevel)}
                                  </div>
                                </div>
                              )}
                            </div>
                          </div>
                          <div className="order-2 my-xxs self-stretch">
                            <div className="printOnly hidden">
                              <img
                                src={favProduct.productImage?.src}
                                alt={favProduct.productImage?.alt}
                                className="mx-auto h-[350px]"
                              />
                            </div>
                            <Link href={favProduct.productDetailPageLink.url ?? ''}>
                              <a className="no-print">
                                <ImagePrimary
                                  imageLayout={'intrinsic'}
                                  fields={{
                                    primaryImageCaption: {
                                      value: '',
                                    },
                                    primaryImage: { value: favProduct.productImage },
                                    primaryImageMobile: { value: favProduct.productImageMobile },
                                    primaryImageMobileFocusArea:
                                      favProduct.primaryImageMobileFocusArea &&
                                      favProduct.primaryImageMobileFocusArea.targetItem.value.value
                                        ? {
                                            id: '',
                                            url: '',
                                            name: favProduct.primaryImageMobileFocusArea.targetItem
                                              .value.value,
                                            displayName:
                                              favProduct.primaryImageMobileFocusArea.targetItem
                                                .value.value,
                                            fields: {
                                              Value: {
                                                value:
                                                  favProduct.primaryImageMobileFocusArea.targetItem
                                                    .value.value,
                                              },
                                            },
                                          }
                                        : {
                                            id: '',
                                            url: '',
                                            name: 'Center',
                                            displayName: 'Center',
                                            fields: {
                                              Value: {
                                                value: 'center',
                                              },
                                            },
                                            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                                            templateName: 'Enum',
                                          },
                                  }}
                                />
                              </a>
                            </Link>
                          </div>
                          <RichTextWrapper
                            field={{ value: favProduct.productDescription.value }}
                            classes={themeData.classes.body}
                          />
                          <div className={themeData.classes.awColorsandCTA}>
                            <ColorAndActions favProduct={favProduct} />
                          </div>
                        </div>
                      </div>
                    );
                  }
                )}
            </div>
          </>
        ) : (
          // No Favorite Product
          <NoResults fields={fields} />
        )}
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<FavoriteProductsProps>(FavoriteProducts);
