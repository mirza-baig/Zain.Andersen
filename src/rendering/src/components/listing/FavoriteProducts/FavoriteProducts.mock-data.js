const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'FavoriteProducts',
  },

  fields: {
    noResultsBodyCopy: {
      value: "Here's some other stuff to try.",
    },
    noResultsHeadline: {
      value: 'Looks like you have no saved products.',
    },
    card1Headline: {
      value: 'No results headline 1',
    },
    card1Subheadline: {
      value: 'No results subheadline 1',
    },
    card2Headline: {
      value: 'No results headline 2',
    },
    card2Subheadline: {
      value: 'No results subheadline 2',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    headlineLevel: {
      id: '7f10288a-beb1-4d95-a139-32df40094441',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3',
      name: 'Heading 3',
      displayName: 'Heading 3',
      fields: {
        Value: {
          value: 'h3',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'Favorite Products',
    },
    primaryImage: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Big-Doors-900x900.jpg?h=900&iar=0&w=900',
        alt: 'Big-Doors-900x900',
        width: '900',
        height: '900',
      },
    },
    primaryImageMobile: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Big-Doors-900x900.jpg?h=900&iar=0&w=900',
        alt: 'Big-Doors-900x900',
        width: '900',
        height: '900',
      },
    },
    primaryImageMobileFocusArea: {
      id: 'e6b36945-ebc0-4af8-ad95-d8268b56b7d0',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Focus/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    secondaryImageMobileFocusArea: {
      id: 'e6b36945-ebc0-4af8-ad95-d8268b56b7d0',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Focus/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    secondaryImage: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Square1ratio1/cass-smith-front-door--blog-392x392.jpg?h=392&iar=0&w=392',
        alt: 'woman in entrance of home with andersen black framed door open',
        width: '392',
        height: '392',
      },
    },
    secondaryImageMobile: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Media/Images/Square1ratio1/cass-smith-front-door--blog-392x392.jpg?h=392&iar=0&w=392',
        alt: 'woman in entrance of home with andersen black framed door open',
        width: '392',
        height: '392',
      },
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Link: {
      value: {
        href: '/Components',
        text: 'Link text 1',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{956E311E-E282-443E-85E8-3B68F03F313E}',
      },
    },
    cta1Modal: null,
    cta1ModalLinkText: {
      value: '',
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Link: {
      value: {
        href: '/Components',
        text: 'Link text 2',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{956E311E-E282-443E-85E8-3B68F03F313E}',
      },
    },
    cta2Modal: null,
    cta2ModalLinkText: {
      value: '',
    },
    cta2Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
