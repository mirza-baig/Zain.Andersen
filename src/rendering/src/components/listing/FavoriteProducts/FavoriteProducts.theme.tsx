// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const FavoriteProductsTheme: ThemeFile = {
  aw: {
    classes: {
      mainheadline: 'font-sans font-medium md:font-heavy text-black text-s md:text-m',
      productPreviewCradWrapper: 'flex flex-col items-start border border-gray h-full relative',
      eyebrow: 'text-dark-gray uppercase font-sans text-xxs font-heavy order-1',
      headlineWrapper: 'order-2 font-heavy font-sans text-xs',
      headline: 'text-xs font-heavy hover:underline',
      ratingsAndPriceWrapper: 'flex items-center order-3',
      ratingsIconsList: 'flex mr-xxs',
      ratingsText: 'pr-xxs text-xs font-sans',
      priceLevelWrapper: 'border-l border-secondary px-xxs ',
      priceText: 'font-sans leading-none flex [&>*:nth-child(2)]:text-gray',
      priceLevel: 'text-xs font-heavy font-sans',
      body: 'px-s my-s order-3 text-dark-gray',
      headerWrapper: 'flex flex-col items-start px-s pb-s pt-m gap-2 order-1',
      awColorsandCTA:
        'flex max-lg:flex-wrap  lg:mb-s lg:items-end justify-between w-full mt-auto max-lg:mb-s order-3',
      colorSwatchesWrapper: 'px-xs max-lg:mb-s',
      colorLabel: 'text-small',
      swatches: 'mt-xxxs flex',
      actions: 'px-xxs md:px-s order-5 max-md:[&_a]:px-xxs [&_a]:mr-0 max-md:mt-auto',
      buttonGroupClass: {
        wrapper: 'mb-0',
        cta1Classes: 'mb-0',
      },
      colorSwatches:
        'h-[30px] w-[30px] rounded-full mr-[14px] [&_*]:h-[30px] [&_*]:max-h-[30px] [&_*]:w-[30px] [&_img]:rounded-full',
      favoriteProduct:
        'favorite-product-item absolute right-0 top-0 inline h-0 w-0 cursor-pointer border-t-0 border-l-0 border-r-[60px] border-b-[60px] border-solid border-[transparent_#e3e3e3_transparent_transparent] transition-[border-color]  duration-500 ease-[ease]',
      favoriteIcon: 'absolute -right-[51px] top-[11px]',
      noResultsHeadline: 'open-sans text-m font-medium',
      cardHeadline: 'open-sans text-s font-heavy',
      cardSubheadline: 'open-sans text-xs font-medium',
    },
  },
  rba: {
    classes: {
      /** Insert Theme classes here **/
    },
  },
};
