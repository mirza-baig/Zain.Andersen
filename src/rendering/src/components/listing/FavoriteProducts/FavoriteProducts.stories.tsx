// Global
import { Story, Meta } from '@storybook/react';
// Local
import FavoriteProducts, { FavoriteProductsProps } from './FavoriteProducts.dynamic';
import defaultData from './FavoriteProducts.mock-data';

export default {
  title: 'Listing/FavoriteProducts',
  component: FavoriteProducts,
} as Meta;
const Template: Story<FavoriteProductsProps> = (props) => <FavoriteProducts {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
