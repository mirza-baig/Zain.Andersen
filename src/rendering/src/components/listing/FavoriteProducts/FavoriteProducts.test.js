// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import FavoriteProducts from './FavoriteProducts.dynamic';
import defaultData from './FavoriteProducts.mock-data';

beforeEach(() => {
  fetch.resetMocks();
});

it('renders correctly', async () => {
  fetch.mockResponseOnce(JSON.stringify({ productData: [] }));

  const component = snapshot(FavoriteProducts, { componentProps: defaultData });
  await hasDataComponent(component, 'listing/favoriteproducts');
});
