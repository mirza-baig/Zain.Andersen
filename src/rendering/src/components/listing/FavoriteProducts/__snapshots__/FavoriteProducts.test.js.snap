// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="listing/favoriteproducts"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="printSection col-span-12 flex justify-between"
      >
        <div
          class="items-top font-sans [&_a:hover]:underline font-sans font-medium md:font-heavy text-black text-s md:text-m"
        >
          <h4>
            Favorite Products
          </h4>
        </div>
      </div>
      <div
        class="col-span-12"
      >
        <div>
          <div>
            <div
              class="font-sans text-m font-medium"
            >
              Looks like you have no saved products.
            </div>
            <div
              class="body-copy"
            >
              <div
                class="mt-[20px]"
              >
                Here's some other stuff to try.
              </div>
            </div>
          </div>
          <div
            class="mt-m grid grid-cols-1 gap-4 ml:grid-cols-2"
          >
            <div>
              <div
                class="relative hidden md:block aspect-picture   "
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="Big-Doors-900x900"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative block w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="Big-Doors-900x900"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="mt-s font-sans text-s font-heavy"
              >
                No results headline 1
              </div>
              <div
                class="font-sans text-xs font-medium"
              >
                No results subheadline 1
              </div>
              <div
                class="mt-m mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  aria-label="Link text 1"
                  class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href="/Components/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Link text 1
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
            <div>
              <div
                class="relative hidden md:block aspect-picture   "
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="woman in entrance of home with andersen black framed door open"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative block w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="woman in entrance of home with andersen black framed door open"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="mt-s font-sans text-s font-heavy"
              >
                No results headline 2
              </div>
              <div
                class="font-sans text-xs font-medium"
              >
                No results subheadline 2
              </div>
              <div
                class="mt-m mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  aria-label="Link text 2"
                  class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href="/Components"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Link text 2
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
